### ISSUE TITLE
[BUG] Weight calculation not based on item quantity

### ISSUE THREAD

**Issue by Rick-flashpoint** [tig-nl/postnl-magento2#348]
**To Reproduce**
1. Have the following matrix:
```
NL,*,*,0.0000,0.0000,0,regular,2.0000,
NL,*,*,5.0000,0.0000,0,regular,15.0000,
NL,*,*,10.0000,0.0000,0,*,20.0000,

```
2. If you have a product with the weight of 5KG, 1 should result in a cost of 15, 2 should result in a cost of 20. But at this point, both is will result in a cost of 10
3. you can see the result if you call this: postnl/deliveryoptions/timeframes/:
```JSON
{"error":"Bezorgdagen opties staan uitgeschakeld.","price":15,"timeframes":[[{"fallback":"2-3 werkdagen levertijd"}]]}
```

The final calculation is correct, but the checkout will show the wrong lower price from the matrix.

**Expected result**
A higher price if items have multiple qty

**Actual result**
Same price in checkout

**Workaround**
change in [Service/Carrier/QuoteToRateRequest.php:](https://github.com/tig-nl/postnl-magento2/blob/7198ade288f8883081749b60a65f9b827d5c5122/Service/Carrier/QuoteToRateRequest.php#L104)
```php
return $item->getWeight();
```
TO
```php
return $item->getRowWeight();
```

**Commit**
The issue is introduced in this commit:
https://github.com/tig-nl/postnl-magento2/commit/a24eb20df64de64d3588a1ce2690655182314d58

** Please complete the following information**
- Magento version:  2.4.5-p1
- PHP version: 8.1
- TIG PostNL version: 1.12.5

**Comment by tig-jeffreybranderhorst** [comment]
Hi @Rick-flashpoint ,

I was able to reproduce this, it happens because the sort order places the NL,*,*,10.0000,0.0000,0,*,20.0000, below the NL,*,*,5.0000,0.0000,0,regular,15.0000, so it stops the moment it finds NL,*,*,5.0000,0.0000,0,regular,15.0000. 

I'm going to place this on our backlog to fix, I will also place your solution there, so we can test your solution with multiple scenarios. Thank you for submitting this issue, it will be on our backlog. 

Have a great day,
Jeffrey

**Comment by tig-vincentthart** [comment]
@Rick-flashpoint 

Thanks again for sharing this fix, we have included it into the latest release v1.12.7.

Greetings

Vincent
